src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGen.scala

```diff
@@ -655,12 +655,19 @@ class CodeGen (val target: Target) {
 
         for (decl <- aContract.declarations) {
             translateDeclaration(decl, newClass, translationContext, aContract)
-            if (decl.isInstanceOf[Constructor] && (!generated)) {
+            if (decl.isInstanceOf[Constructor] && (!generated) && aContract.modifiers.contains(IsMain())) {
```

Comment:
Could you please add an isMain() convenience method to Contract? There's already an isResource() method to model it on.